src/theory/quantifiers/sygus/synth_conjecture.h

```diff
@@ -165,6 +162,8 @@ class SynthConjecture
   TermDbSygus* d_tds;
   /** The feasible guard. */
   Node d_feasible_guard;
+  /** do we have a solution in this user context? */
```

Comment:
please say more why it's necessary to keep this tracking at the level of the user context (which is not done for anything else in the module)

Reply:
I've documented, note this flag is in preparation of sygus + incremental (it will be clearer on future PRs).